ophyd/signal.py

```diff
@@ -839,11 +839,34 @@ def get(self, *, as_string=None, connection_timeout=1.0, form='time',
             as_string = self._string
 
         self.wait_for_connection(timeout=connection_timeout)
-        info = self._read_pv.get_with_metadata(as_string=as_string, form=form,
-                                               **kwargs)
+
+        def restorewarnings(filters):
+            warnings.resetwarnings()
+            for f in filters:
+                action = f[0]
+                message = f[1] or ""
+                category = f[2]
+                module = f[3] or ""
+                lineno = f[4]
+                warnings.filterwarnings(action, message, category, 
+                                        module, lineno, append=True) 
+
+        old_warning_filters = list(warnings.filters)
+        warnings.filterwarnings("error")
```

Comment:
This should probably be in the `try` just to be safe.  `KeyboardInterupt` can be raised at any point of the code (python checks to see if the signal has been sent between every byte code) so it is possible for this to raise _between_ the state in warnings being mutated.

Reply:
So, going with the return-value based detection of timeout, we don't need `try..except..finally`.  I agree with the *less code to be run* approach.  The comments here should be added to the code to avoid repeat of this suggested *improvement*.

We'll need to take control of the `timeout` value, including the logic of scaling it with the array `count`.  We can't capture any of the message posted by PyEpics to `warnings.warn()` and remain thread-safe.